pkg/network/node/node.go

```diff
@@ -311,38 +311,36 @@ func (node *OsdnNode) validateMTU() error {
 		return fmt.Errorf("unable to determine MTU while performing validation")
 	}
 
-	if int(node.networkInfo.MTU) > mtu+50 {
+	const MTUTaintKey string = "network.openshift.io/mtu-too-small"
+	n, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
+	}
+	mtuTooSmallTaint := &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"}
+	tainted := taints.TaintExists(n.Spec.Taints, mtuTooSmallTaint)
+	mtuWithOverhead := mtu + 50
+	if !tainted && int(node.networkInfo.MTU) > mtuWithOverhead {
 		resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 			klog.V(2).Infof("Configured node MTU is more than default interface MTU plus VXLAN overhead, tainting node...")
-			n, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
```

Comment:
Oh, actually you need to keep the `Get` in the loop even if you already did it outside the loop. The point of `RetryOnConflict` is that if two people try to update at once, the first one will succeed, and that will change the `resourceVersion` on the object, and then when the the apiserver tries to process the second update, it will see that the `resourceVersion` you passed in is old and so return `409 Conflict` and force you to try again so that you don't accidentally undo the changes that the first person made. So anyway, this means that if you get a conflict, you need to call `Get` again to get the newly-updated version of the resource before trying to update it again.

tl;dr: keep the `Get`. And add it to the "remove" case.

Actually, you could merge a lot of the "add" and "remove" cases together by just testing if `tainted != needsTaint` first, and if so, then do the `RetryOnConflict` loop, and then inside that, figure out whether you need to be calling `AddOrUpdateTaint` or `RemoveTaint`, and everything else would be the same.